src/renate/shift/mmd_detectors.py

```diff
@@ -0,0 +1,45 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Optional
+
+import torch
+
+from renate.shift.detector import ShiftDetectorWithFeatureExtractor
+from renate.shift.mmd_helpers import mmd, RBFKernel
+
+
+class MMDCovariateShiftDetector(ShiftDetectorWithFeatureExtractor):
+    """A kernel maximum mean discrepancy (MMD) test.
+
+    This test was proposed by
+
+    [1] Gretton, A., et al. A kernel two-sample test. JMLR (2012).
+
+    We currently do not expose the choice of kernel. It defaults to an RBF kernel with a lengthscale
+    set via the median heuristic.
+
+    Args:
+        feature_extractor: A pytorch model used as feature extractor.
+        num_permutations: Number of permutations for permutation test.
+        batch_size: Batch size used to iterate over datasets.
+        num_preprocessing_workers: Number of workers used in data loaders.
+        device: Device to use for computations inside the detector.
+    """
+
+    def __init__(
```

Comment:
I think that exposing the choice of the kernel to be used could be beneficial in the long term

Reply:
Agree, but I'd like to move along with this quickly. I think we can easily integrate other kernels later.

Reply:
OK, please add a TODO

Reply:
Added an issue.